### PR TITLE
VZ-5680: Allow VMO to install without hard dependencies

### PR DESCRIPTION
**Pull request by bobdonat**
# Description

This pull request allows VMO to be installed if its dependency on nginx is not enabled.  This allows for only VMO/OS/OSD to be installed. 

Dependencies during install are now honored only if the dependency is enabled and the webhook validation to check if dependencies are enabled was removed. 

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

**Review by anders-swanson — APPROVED** [review]

File: platform-operator/helm_config/charts/verrazzano-monitoring-operator/templates/configmap.yaml [hunk 1]
```diff
@@ -13,4 +13,8 @@ data:
     metricsPort: {{ .Values.monitoringOperator.metricsPort }}
     defaultSimpleCompReplicas: {{ .Values.monitoringOperator.defaultSimpleCompReplicas }}
     defaultPrometheusReplicas: {{ .Values.monitoringOperator.defaultPrometheusReplicas }}
+    {{- if .Values.config.envName }}
+    {{- if .Values.config.dnsSuffix }}
     defaultIngressTargetDNSName: {{ .Values.config.envName }}-verrazzano-ingress.{{ .Values.config.dnsSuffix }}
```
Comment (anders-swanson) [inline platform-operator/helm_config/charts/verrazzano-monitoring-operator/templates/configmap.yaml:18]: is this value used?

Comment (bobdonat) [reply platform-operator/helm_config/charts/verrazzano-monitoring-operator/templates/configmap.yaml:18]: VMO uses it when
  vmo.Spec.IngressTargetDNSName == ""

Comment (bobdonat) [reply platform-operator/helm_config/charts/verrazzano-monitoring-operator/templates/configmap.yaml:18]: maybe that code is not needed in VMO

**Review by michael-cico — APPROVED** [review]

**Comment by michael-cico** [comment]
Will it work standalone without Keycloak?  What is used for auth?

**Comment by bobdonat** [comment]
> Will it work standalone without Keycloak? What is used for auth?

As I understand the idea was that keycloak/auth proxy would not be setup and auth setup was an exercise for the user

**Comment by mbenmclean** [comment]
So there will need to be a subsequent change to add the context to the GetDependencies function and add logic to create optional dependencies, right. Because this change on it's own won't allow the VMI components to get installed without NGINX. They will still try to wait for it.

**Comment by bobdonat** [comment]
> So there will need to be a subsequent change to add the context to the GetDependencies function and add logic to create optional dependencies, right. Because this change on it's own won't allow the VMI components to get installed without NGINX. They will still try to wait for it.

With this change a component will not wait for a component if it is not enabled.  We will have to test the VMI components to see if there are any other changes needed.

**Pull request merged**